sihl/src/web_flash.ml

```diff
@@ -103,28 +100,36 @@ let decode_flash cookie_key req =
             "Maybe the cookie key '%s' collides with a cookie issued by \
              someone else. Try to change the cookie key."
             cookie_key);
-      Flash.empty
-    | Some flash -> flash)
+      Parse_error
+    | Some flash -> Found flash)
 ;;
 
-let persist_flash old_flash cookie_key resp =
+let persist_flash ?old_flash ?(delete_if_not_set = false) cookie_key resp =
   let flash = Opium.Context.find Env.key resp.Opium.Response.env in
   match flash with
+  (* No flash was set in handler *)
   | None ->
-    if Flash.is_empty old_flash
-    then (* Flash was not touched, don't set cookie *)
-      resp
-    else (* Remove flash cookie *)
+    if delete_if_not_set
+    then
+      (* Flash was not touched, don't set cookie *)
       Opium.Response.remove_cookie cookie_key resp
-  | Some flash ->
-    if Flash.equals old_flash flash
-    then (* Flash was not touched, don't set cookie *)
+    else (* Remove flash cookie *)
```

Comment:
This comment should be for the `then` branch and vice versa, correct?

Reply:
Fixed